File: internal/content/testdata/bench/csharp/zigzag-string-read.cs
using System.Text;

public class Solution {
    public string zigzag_convert(string s, long num_rows) {
        if (num_rows == 1) return s;
        var rows = new StringBuilder[num_rows];
        for (int i = 0; i < num_rows; i++) rows[i] = new StringBuilder();
        int row = 0, direction = -1;
        foreach (char ch in s) {
            rows[row].Append(ch);
            if (row == 0 || row == (int)num_rows - 1) {
                direction = -direction;
            }
            row += direction;
        }
        var sb = new StringBuilder();
        foreach (var r in rows) sb.Append(r);
        return sb.ToString();
    }
}

File: internal/content/testdata/bench/java/zigzag-string-read.java
class Solution {
    public String zigzag_convert(String s, long num_rows) {
        int numRows = (int) num_rows;
        if (numRows == 1) return s;
        StringBuilder[] rows = new StringBuilder[numRows];
        for (int i = 0; i < numRows; i++) rows[i] = new StringBuilder();
        int row = 0, direction = -1;
        for (char ch : s.toCharArray()) {
            rows[row].append(ch);
            if (row == 0 || row == numRows - 1) direction = -direction;
            row += direction;
        }
        StringBuilder result = new StringBuilder();
        for (StringBuilder r : rows) result.append(r);
        return result.toString();
    }
}
